gobblin-data-management/src/main/java/gobblin/data/management/copy/hive/HiveDataset.java

```diff
@@ -119,244 +83,52 @@ public HiveDataset(FileSystem fs, HiveMetastoreClientPool clientPool, Table tabl
-    }
-
-    return copyEntities;
-
-  }
-
-  private List<CopyEntity> getCopyEntitiesForUnpartitionedTable(FileSystem targetFs,
-      Optional<Table> targetTable, CopyConfiguration configuration) throws IOException {
-
-    List<CopyEntity> copyEntities = Lists.newArrayList();
-
-    InputFormat<?, ?> inputFormat = getInputFormat(this.table.getSd());
-
-    Collection<Path> sourcePaths = getPaths(inputFormat, this.table.getSd().getLocation());
-    Collection<Path> targetExistingPaths = targetTable.isPresent() ?
-        getPaths(inputFormat, targetTable.get().getSd().getLocation()) : Sets.<Path>newHashSet();
-
-    DiffPathSet diffPathSet = diffSourceAndTargetPaths(sourcePaths, targetExistingPaths,
-        Optional.<Partition>absent(), targetFs);
-
-    // TODO: create delete work units for files that must be deleted on target
-
-    for (CopyableFile.Builder builder : getCopyableFilesFromPaths(diffPathSet.filesToCopy,
-        configuration, Optional.<Partition>absent())) {
-      copyEntities.add(builder.build());
-    }
-
-    return copyEntities;
-  }
-
-  private DiffPathSet diffSourceAndTargetPaths(Collection<Path> sourcePaths, Collection<Path> targetExistingPaths,
-      Optional<Partition> partition, FileSystem targetFs) throws IOException {
-    DiffPathSet.DiffPathSetBuilder builder = DiffPathSet.builder();
-
-    Set<Path> targetPathSet = Sets.newHashSet(targetExistingPaths);
-
-    for (Path sourcePath : sourcePaths) {
-      FileStatus status = this.fs.getFileStatus(sourcePath);
-      Path targetPath = targetFs.makeQualified(getTargetPath(status, partition));
-      if (targetPathSet.contains(targetPath)) {
-        FileStatus targetStatus = targetFs.getFileStatus(targetPath);
-        if (targetStatus.getLen() != status.getLen() || targetStatus.getModificationTime() < status.getModificationTime()) {
-          // We must replace target file
-          builder.deleteFile(targetPath);
-          builder.copyFile(status);
-        }
-        targetPathSet.remove(targetPath);
-      } else {
-        builder.copyFile(status);
-      }
-    }
-
-    // All leftover paths in target are not in source, should be deleted.
-    for (Path deletePath : targetPathSet) {
-      builder.deleteFile(deletePath);
+    try {
+      return new HiveCopyEntityHelper(this, configuration, targetFs).getCopyEntities();
+    } catch (IOException ioe) {
+      log.error("Failed to copy table " + this.table, ioe);
+      return Lists.newArrayList();
     }
-
-    return builder.build();
-  }
-
-  @Builder
-  private static class DiffPathSet {
-    @Singular(value = "copyFile") Collection<FileStatus> filesToCopy;
-    @Singular(value = "deleteFile") Collection<Path> pathsToDelete;
   }
 
-  private Map<List<String>, Partition> getPartitions(AutoReturnableObject<IMetaStoreClient> client, Table table)
+  protected static Map<List<String>, Partition> getPartitions(AutoReturnableObject<IMetaStoreClient> client, Table table)
```

Comment:
Why not moving these `static` methods to `utils`? They are generic and could be useful for others.